lib/matplotlib/colorbar.py

```diff
@@ -1150,7 +1150,7 @@ def _mesh(self):
 
     def _forward_boundaries(self, x):
         b = self._boundaries
-        y = np.interp(x, b, np.linspace(0, b[-1], len(b)))
+        y = np.interp(x, b, np.linspace(0, np.abs(b[-1]), len(b)))
```

Comment:
Can you clarify what is being computed here?  Without context, it seems wrong that only the last point of `b` is taken into account, in particular the example in #21671 fails again (with this PR) if `b[-1] == 0` (e.g., `clevs = list(range(-94, 1))`), or rather the resulting plot is wrong.

Reply:
Good catch.  Fixed.  OTOH do we know if `clevs=np.arange(1, -9, -1)` should be allowed?

Reply:
I would like decreasing levels to be allowed, but to be clear that's a separate issue that can be dealt with later; the problem here is with increasing levels where the last level is zero.

Reply:
Yes, I can confirm that `clev=np.arange(1, -9, -1)` used to return nonsense on v3.4.3.  It errors on this PR as of my latest push, which seems better than nonsense, but it can probably be made to work.

Reply:
Would it work to just set the second arg to `range(len(b))` instead?

Reply:
You mean the third arg?

`y = np.interp(x, b, np.arange(len(b)))`

That works as well.

Reply:
The only issue is that there is this 1e-6 scaling in there, that doesn't seem tested, but was apparently there for a reason, so using the actual spread in the boundaries gives a natural scaling for that...

Reply:
though now that I look at it ` y[x > b[-1]+eps] = 2` can't possibly be right...

Reply:
Yes, I meant the third.  I'm also a bit confused by the setting to -1/+2 for out-of-bounds values...

Reply:
No thats clearly wrong.  Fixed in the latest.  However, I played with this, and there are no tests of this happening, so I'm not sure where this hedging came from.  I _think_ it may have been extra ticks in the extend region, but that can't really happen (because its just bounds mapping to bounds) so maybe we don't need all this...

Reply:
I see that in the new version you're mapping to `np.linspace(0, db, len(b))`, which, while not "wrong", still seems weird to me (I still think the mapped-to space should be either `range(len(b))` or `np.linspace(0, 1, len(b))` -- `db` doesn't have any particular meaning in that space).

Reply:
OK, I think that works...